tests/ha/test_multi_server_pod_failure.py

```diff
@@ -285,3 +289,330 @@ def test_reads_kserver_pods_fail(self):
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs with variable sizes objects.")
+
+        LOGGER.info("Step 2: Perform READs and verify DI on the written data")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 2: Performed READs and verified DI on the written data")
+
+        LOGGER.info("Shutdown %s (K) server pods one by one and verify read/verify "
+                    "after each pod down", self.kvalue)
+        LOGGER.info("Get server pod names to be deleted")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.SERVER_POD_NAME_PREFIX)
+        self.pod_name_list = random.sample(pod_list, self.kvalue)
+        for count, pod_name in enumerate(self.pod_name_list):
+            count += 1
+            LOGGER.info("Step 3: Shutdown %s server pod %s by deleting deployment (unsafe)",
+                        count, pod_name)
+            pod_data = list()
+            pod_data.append(
+                self.node_master_list[0].get_pod_hostname(pod_name=pod_name))  # hostname
+            resp = self.node_master_list[0].delete_deployment(pod_name=pod_name)
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_false(resp[0], f"Failed to delete {count} server pod {pod_name} "
+                                               "by deleting deployment (unsafe)")
+            pod_data.append(resp[1])  # deployment_backup
+            pod_data.append(resp[2])  # deployment_name
+            self.pod_dict[pod_name] = pod_data
+            LOGGER.info("Step 3: Deleted %s server pod %s by deleting deployment (unsafe)",
+                        count, pod_name)
+
+            LOGGER.info("Step 4: Check cluster status")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 4: Cluster has some failures due to %s server pods has gone down.",
+                        count)
+
+            LOGGER.info("Step 5: Check services status that were running on server pod %s which "
+                        "are deleted", pod_name)
+            hostname = self.pod_dict.get(pod_name)[0]
+            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
+                                                               hostname=hostname)
+            assert_utils.assert_true(resp[0], resp[1])
+            pod_list.remove(pod_name)
+            LOGGER.info("Step 5: Services of server pod %s are in offline state", pod_name)
+
+            LOGGER.info("Step 6: Check services status on remaining pods %s", pod_list)
```

Comment:
do we need to check on data pods service too ? @sonalk0209  @ApurwaMohite, did we missed it ?